sops.go

```diff
@@ -70,6 +70,11 @@ const MacMismatch = sopsError("MAC mismatch")
 // MetadataNotFound occurs when the input file is malformed and doesn't have sops metadata in it
 const MetadataNotFound = sopsError("sops metadata not found")
 
+// MACOnlyEncryptedInitialization is a constant and known sequence of 32 bytes used to initialize
+// MAC which is computed only over values which end up encrypted. That assures that a MAC with the
+// setting enabled is always different from a MAC with this setting disabled.
+var MACOnlyEncryptedInitialization = []byte{230, 33, 177, 209, 12, 4, 122, 97, 127, 148, 55, 76, 77, 172, 154, 145, 62, 252, 196, 118, 70, 250, 45, 185, 212, 142, 78, 165, 226, 1, 23, 126}
```

Comment:
Where do these numbers come from? I would prefer something that is determinstic (like a string of digits from pi or e or some other known constant).

Reply:
I just generated a random UUIDv4. It does not really matter what it is. It also cannot really influence the correctness/security of crypto here (like curve constants).

So feel free to propose any other sequence of bytes. :-)

Reply:
Just to be clear, personally, I would vote against changing this. It really does not matter so much what it is but some people have been using this branch already (including me) so it would be just extra work for them to re-encrypt stuff to recompute MACs with another constant.

But I can change it to something else if you insist. Just say what.

Reply:
It's a generally accepted principle in cryptography that you should avoid any kind of magic number, see https://en.wikipedia.org/wiki/Nothing-up-my-sleeve_number. So this really should be changed.

That said, the value doesn't really matter. It could be something regular, like my first suggestion below, or something random-looking, like my second suggestion.

Something regular:
```suggestion
var MACOnlyEncryptedInitialization = []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32}
```

Something random looking:
```suggestion
// The following numbers are taken from the output of `echo -n sops | sha256sum` (shell) or `hashlib.sha256(b'sops').hexdigest()` (Python).
var MACOnlyEncryptedInitialization = []byte{0x8a, 0x3f, 0xd2, 0xad, 0x54, 0xce, 0x66, 0x52, 0x7b, 0x10, 0x34, 0xf3, 0xd1, 0x47, 0xbe, 0xb, 0xb, 0x97, 0x5b, 0x3b, 0xf4, 0x4f, 0x72, 0xc6, 0xfd, 0xad, 0xec, 0x81, 0x76, 0xf2, 0x7d, 0x69}
```

Reply:
Thanks for the explanation. I picked the second one.